README.md

```diff
@@ -1,120 +1,237 @@
-serialize_model(
-    model=pipeline.vae,
-    config=None,
-    model_directory=model_id,
-    model_prefix="vae",
-)
+```python
+  TensorDeserializer(
+     open_stream(s3_uri,
+                 "rb",
+                 aws_access_key_id=AWS_ACCESS_KEY,
+                 aws_secret_access_key=AWS_SECRET_KEY,
+                 s3_endpoint="object.ord1.coreweave.com"))
+```
 
-# The third model is the UNet.
-serialize_model(
-    model=pipeline.unet,
-    config=None,
-    model_directory=model_id,
-    model_prefix="unet",
-)
+## Additional Features
+`tensorizer` has a few additional features that make it more useful than
+just a serialization/deserialization tool.
 
-# This is optional, but you can also save the CLIP tokenizer and the Stable
-# Diffusion scheduler to the model directory.
-pipeline.tokenizer.save_pretrained(model_id)
+### Plaid Mode
+`tensorizer` has a `plaid_mode` argument that can be passed to the
+`TensorDeserializer` class. When `plaid_mode` is `True`, `tensorizer`
+will load the tensors extremely fast. This is done by loading the tensors
+into a `torch.nn.Module` that is not initialized, by overriding the
+`__init__` method of the `torch.nn.Module` to do nothing.
 
-# We can also save the scheduler.
-pipeline.scheduler.save_config(model_id)
+The tensors are them loaded into a buffer, and the buffer is zero-copied
+into the uninitialized `torch.nn.Module`. This is unsafe, and should only
+be used in inference cases where the model is not being trained.
 
-# To load the models into a blank Stable Diffusion pipeline, we have to import
-# the individual models first so we can initialize the pipeline with them.
+### `state_dict` Support
+The `TensorDeserializer` object can be used as-is as a `state_dict` for
+`torch.nn.Module.load_state_dict`. This is useful for loading the tensors
+into a `torch.nn.Module` that is already initialized, or for inspection.
 
-from diffusers import AutoencoderKL, UNet2DConditionModel, DDIMScheduler
-from transformers import CLIPTextModel, CLIPTextConfig, CLIPTokenizer
+Keep in mind that `load_state_dict` is not a fast operation, and will
+likely be much slower than `load_into_module`.
 
-vae = load_model(model_id, AutoencoderKL, None, "vae")
-unet = load_model(model_id, UNet2DConditionModel, None, "unet")
-encoder = load_model(model_id, CLIPTextModel, CLIPTextConfig, "encoder")
+The `state_dict` can also be used to initialize a HuggingFace Transformers
+AutoModel. But HuggingFace Transformers performs three or more copies of
+the data, so memory use will explode.
 
-pipeline = StableDiffusionPipeline(
-    text_encoder=encoder,
-    vae=vae,
-    unet=unet,
-    scheduler=DDIMScheduler.from_config(model_id),
-    tokenizer=CLIPTokenizer.from_pretrained(model_id),
-    safety_checker=None,
-    feature_extractor=None,
-)
+## Running Tests
+`tensorizer` uses `pytest` for testing. It has its own set of dependencies,
+which can be installed with `pip install -r tests/requirements.txt`.
```

Comment:
Changed 👍

As a small note we could choose the first step to be either `python -m pip install -e .` or `python -m pip install -r requirements.txt`, since we have both a `pyproject.toml` and a `requirements.txt`. The former also installs `tensorizer` itself in editable mode and makes it globally available to import, whereas the latter would only install dependencies. The difference would only be important if there were some conflicts installing versus an already installed version of `tensorizer`, I think.